doc/source/ray-core/ray-dashboard.rst

```diff
@@ -6,74 +6,297 @@ Ray provides a web-based dashboard for monitoring and debugging Ray applications
+- :ref:`See the progress of your job <dash-workflow-job-progress>`.
+- :ref:`Find the application logs or error messages of failed tasks or actors <dash-workflow-logs>`.
+- :ref:`Profile, trace dump, and visualize the timeline of the Ray jobs, tasks, or actors <dashboard-profiling>`.
+- :ref:`Analyze the CPU and memory usage of the cluster, tasks and actors <dash-workflow-cpu-memory-analysis>`.
+- :ref:`See individual state of task, actor, placement group <dash-workflow-state-apis>`, and :ref:`nodes (machines) <dash-node-view>` which is equivalent to :ref:`Ray state APIs <state-api-overview-ref>`.
+- :ref:`View the hardware utilization (CPU, GPU, memory) <dash-workflow-resource-utilization>` so that you can compare it to Ray's logical resource usages.
+
 Getting Started
 ---------------
 
-To use the dashboard, first install Ray with the proper dependencies:
+To use the dashboard, you should use the `ray[default]` installation:
 
 .. code-block:: bash
 
   pip install -U "ray[default]"
 
-You can access the dashboard through a URL printed when Ray is initialized (the default URL is **http://localhost:8265**).
+You can access the dashboard through a URL printed when Ray is initialized (the default URL is **http://localhost:8265**) or via the context object returned from `ray.init`.
 
-If you prefer to explicitly set the port on which the dashboard will run, you can pass
-the ``--dashboard-port`` argument with ``ray start`` in the command line, or you can pass the
-keyword argument ``dashboard_port`` in your call to ``ray.init()``.
+.. code-block:: python
+
+    context = ray.init()
+    print(context.dashboard_url)
 
 .. code-block:: text
 
   INFO worker.py:1487 -- Connected to Ray cluster. View the dashboard at 127.0.0.1:8265.
 
-The dashboard is also available :ref:`when using the cluster launcher <monitor-cluster>`.
+Ray cluster comes with the dashboard. See :ref:`Cluster Monitoring <monitor-cluster>` for more details.
 
-Views
------
+.. note:: 
 
-There are 4 different views that the Dashboard provides: :ref:`Node view <dash-node-view>`,
-:ref:`Jobs view <dash-jobs-view>`, :ref:`Actors view <dash-actors-view>`, and :ref:`Logs view <dash-logs-view>`.
+  When using the Ray dashboard, it is highly recommended to also set up Prometheus and Grafana. 
+  Ray dashboard has a tight integration with these tools, and core features such as :ref:`Metrics View <dash-metrics-view>` won't work without them.
 
-.. image:: https://raw.githubusercontent.com/ray-project/Images/master/docs/new-dashboard/overview.png
-    :align: center
-    :width: 20%
+See :ref:`Ray Metrics <ray-metrics>` to learn how to set up Prometheus and Grafana.
 
-.. _dash-node-view:
+If you need to customize the port on which the dashboard will run, you can pass
+the ``--dashboard-port`` argument with ``ray start`` in the command line, or you can pass the
+keyword argument ``dashboard_port`` in your call to ``ray.init()``.
```

Comment:
There are two issues:
1. The ways to customize the ports on VM cluster and [KubeRay](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/config.html#specifying-non-default-ports) clusters are different. Maybe we should link users to the cluster page where we list hwo to customize the ports in different ways.

2. This only works when you create a single-node ray cluster (when there is no existing ray cluster on the machine)?

> or you can pass the keyword argument ``dashboard_port`` in your call to ``ray.init()``